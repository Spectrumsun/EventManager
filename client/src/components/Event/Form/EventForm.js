import React from 'react';
import PropTypes from 'prop-types';
import PickCenter from '../../Center/PickCenter';

const eventForm = props => (
  <div className="card-body" >
    <form onSubmit={props.onSubmit}>
      <div className="form-row">
        <div className="form-group col-md-12">
          <h5><label htmlFor="inputEmail4">Event Name</label></h5>
          <input
            type="text"
            value={props.name}
            onChange={props.onChange}
            name="name"
            className="form-control form-control-lg"
          />
        </div>

        <div className="form-group col-md-6">
          <h5><label htmlFor="inputPassword4">Date</label></h5>
          <input
            type="date"
            value={props.date}
            onChange={props.onChange}
            name="date"
            className="form-control form-control-lg"
          />
        </div>

        <div className="form-group col-md-6">
          <h5><label htmlFor="inputPassword4">Time</label></h5>
          <input
            type="time"
            value={props.time}
            onChange={props.onChange}
            name="time"
            className="form-control form-control-lg"

          />
        </div>
      </div>

      <div className="form-group">
        <h5><label htmlFor="exampleFormControlTextarea1">purpose</label></h5>
        <textarea
          className="form-control"
          value={props.purpose}
          onChange={props.onChange}
          name="purpose"
          rows="3"
          placeholder="Tell people more about the event"
        />
      </div>


      <button
        type="button"
        className="btn btn-primary"
        data-toggle="modal"
        data-target=".bd-example-modal-lg"
      >
             Select Center
      </button>

      <div
        className="modal fade bd-example-modal-lg"
        tabIndex="-1"
        role="dialog"
        aria-labelledby="myLargeModalLabel"
        aria-hidden="true"
      >

        <div className="modal-dialog modal-lg">
          <div className="modal-content">
            <PickCenter selectCenter={id => props.selectCenter(id)} />
          </div>
        </div>
      </div>
      <br />
      <br />
      <button type="submit" className="btn btn-primary btn-lg">Submit</button>
    </form>
  </div>

);

eventForm.propTypes = {
  selectCenter: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  purpose: PropTypes.string.isRequired,
};


export default eventForm;

