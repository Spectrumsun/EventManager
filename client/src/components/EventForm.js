import React, { Component } from 'react';
// import { Link } from 'react-router-dom';

class EventForm extends Component {
  onSubmit(e) {
    e.preventDefault();
    //customHistory.push('/events');
  }

  render() {
    return (
      <div className="card-body" >
        <form onSubmit={this.onSubmit}>
          <div className="form-row">
            <div className="form-group col-md-6">
              <h5><label htmlFor="inputEmail4">Event Name</label></h5>
              <input type="text" className="form-control form-control-lg" required />
            </div>
            <div className="form-group col-md-6">
              <h5><label htmlFor="inputPassword4">Date</label></h5>
              <input type="date" className="form-control form-control-lg" required />
            </div>
            <div className="form-group col-md-6">
              <h5><label htmlFor="inputPassword4">Time</label></h5>
              <input type="time" className="form-control form-control-lg" required />
            </div>
          </div>
          <div className="form-group">
            <h5><label htmlFor="exampleFormControlTextarea1">Description</label></h5>
            <textarea className="form-control" id="exampleFormControlTextarea1" rows="3" placeholder="Tell people more about the event" />
          </div>
          <button type="button" className="btn btn-dark btn-sm" data-toggle="modal" data-target="#exampleModal">
			  	    Pick center
          </button>
          <div className="modal fade" id="exampleModal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
            <div className="modal-dialog " role="document">
              <div className="modal-content">
                <div className="modal-header centerlist">
                  <h5 className="modal-title " id="exampleModalLabel">List of centers</h5>
                  <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                    <span aria-hidden="true">&times;</span>
                  </button>
                </div>
                <div className="modal-body centerlist ">
                  <h3 className="dark color">Havilah Event Centre</h3>
                  <h5 className="center">No 33 Ignobi Street Lagos</h5>
                  <h6 className="center">Facility Available</h6>
                  <ul className="list-group">
                    <li className="list-group-item centerlist" >Open Field</li>
                    <li className="list-group-item centerlist">Sound System</li>
                    <li className="list-group-item centerlist">Free Wifi Hotspot</li>
                    <li className="list-group-item centerlist">Swimming Pool</li>
                  </ul>
                  <br />
                  <button type="button" className="btn btn-primary" data-dismiss="modal">Select</button>
                </div>
                <br />
              </div>
            </div>
          </div>
          <br />
          <br />
          <button type="submit" className="btn btn-primary btn-lg">Submit</button>
        </form>
      </div>

    );
  }
}

export default EventForm;
