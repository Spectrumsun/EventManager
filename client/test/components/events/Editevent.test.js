import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import ConnectedEditEvent,
{ EditEvent, mapDispatchToProps } from '../../../src/components/Event/EditEvent';
import EventForm from '../../../src/components/Event/Form/EventForm'


const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  centers: {
    center: [
      {
        id: 29,
        centerName: 'farm center',
        city: 'Yaba',
        address: 'No 33, yaba road lagos',
        availability: 'yes',
        imageurl: 'llal',
        imageId: 'lol',
        about: 'lalalla',
        facility: ['car', 'fan', 'video']
      },
      {
        id: 9,
        centerName: 'farm center',
        city: 'Yaba',
        address: 'No 33, yaba road lagos',
        availability: 'yes',
        imageurl: 'llal',
        imageId: 'lol',
        about: 'lalalla',
        facility: ['car', 'fan', 'video']
      }
    ],
    pagination: {
      message: 'success',
      result: [
        {
          id: 29,
          centerName: 'farm center',
          city: 'Yaba',
          address: 'No 33, yaba road lagos',
          availability: 'yes',
          imageurl: 'llal',
          imageId: 'lol',
          about: 'lalalla',
          facility: ['car', 'fan', 'video']
        },
        {
          id: 9,
          centerName: 'farm center',
          city: 'Yaba',
          address: 'No 33, yaba road lagos',
          availability: 'yes',
          imageurl: 'llal',
          imageId: 'lol',
          about: 'lalalla',
          facility: ['car', 'fan', 'video']
        }
      ],
      count: 10,
      pages: 2
    },
    error: false,
    centerName: 'farm center',
  },
  events: {
    loadedEvent: {
      eventName: 'tom',
      startDate: '2018-10-20',
      endDate: '2018-10-23',
      time: '10:00',
      purpose: 'for test',
      center: 1,
    }
  },
  match: {
    params: {
      id: 1
    }
  }
};

const store = mockStore(initialState);

const props = {
  onInitCenters: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  onOneEvent: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  initEditEvent: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  page: {
    pagination: {
      message: 'success',
      result: [
        {
          id: 29,
          centerName: 'farm center',
          city: 'Yaba',
          address: 'No 33, yaba road lagos',
          availability: 'yes',
          imageurl: 'llal',
          imageId: 'lol',
          about: 'lalalla',
          facility: ['car', 'fan', 'video']
        },
        {
          id: 9,
          centerName: 'farm center',
          city: 'Yaba',
          address: 'No 33, yaba road lagos',
          availability: 'yes',
          imageurl: 'llal',
          imageId: 'lol',
          about: 'lalalla',
          facility: ['car', 'fan', 'video']
        }
      ],
      count: 10,
      pages: 2
    }
  },
  loadedEvent: {
    eventName: 'tom',
    startDate: '2018-10-20',
    endDate: '2018-10-23',
    time: '10:00',
    purpose: 'for test',
    center: 1,
  },
  match: {
    params: {
      id: '1'
    }
  },
  history: createMemoryHistory(),
  name: 'tom'
};


const shallowWrapper = shallow(<EditEvent {...props} />);

const state = {
  name: 'tomato',
  startDate: '2018-12-05',
  endDate: '2018-06-21',
  time: '21:00',
  purpose: 'fun',
  centerId: 1,
  totalPage: '1',
  next: 1,
  centerName: 'yaba center',
  pageNumber: '2',
  formValid: false,
};

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'tomato',
    startDate: '2018-12-05',
    endDate: '2018-06-21',
    time: '21:00',
    purpose: 'fun',
    centerId: 1,
    totalPage: '1',
    next: 1,
    centerName: 'yaba center',
    pageNumber: '2',
    formValid: false,
  }
};

// let wrapper;

describe('<EditEvent /> Component', () => {
  const wrapper = shallow(<EditEvent {...props} />);
  it('should render the <EditEvent />', () => {
    shallow(<EditEvent {...props} />);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedEditEvent {...props} />
        </BrowserRouter>
      </Provider>);
    expect(tree).toMatchSnapshot();
  });

  it('should render initial layout of <EditEvent />', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should have three div on layout', () => {
    expect(wrapper.find('div').length).toEqual(3);
  });

  it('should have <EventForm /> when the page loads', () => {
    expect(wrapper.find(EventForm)).toHaveLength(1);
  });

  it('calls onChange event when input is passed to state', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event when submit button is clicked', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('calls add event when the previous button is clicked for pagination', () => {
    sinon.spy(shallowWrapper.instance(), 'add');
    shallowWrapper.setState(state);
    shallowWrapper.instance().add();
    expect(shallowWrapper.instance().add.calledOnce).toEqual(true);
  });

  it('calls minus event when the previous button is clicked for pagination', () => {
    sinon.spy(shallowWrapper.instance(), 'minus');
    shallowWrapper.setState(state);
    shallowWrapper.instance().minus();
    expect(shallowWrapper.instance().minus.calledOnce).toEqual(true);
  });

  it('calls getCenter event to load the list of centers', () => {
    sinon.spy(shallowWrapper.instance(), 'getCenter');
    shallowWrapper.setState(state);
    shallowWrapper.instance().getCenter();
    expect(shallowWrapper.instance().getCenter.calledOnce).toEqual(true);
  });

  it('calls selectCenter when center id is selected in when adding editing event', () => {
    sinon.spy(shallowWrapper.instance(), 'selectCenter');
    shallowWrapper.setState(state);
    shallowWrapper.instance().selectCenter();
    expect(shallowWrapper.instance().selectCenter.calledOnce).toEqual(true);
  });

  it('should have three div element match snap', () => {
    expect(wrapper.getElements('div')).toMatchSnapshot();
  });

  it('should have three div element', () => {
    expect(wrapper.find('div').length).toEqual(3);
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).onOneEvent).toBeTruthy();
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).initEditEvent).toBeTruthy();
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).onInitCenters).toBeTruthy();
  });
});

