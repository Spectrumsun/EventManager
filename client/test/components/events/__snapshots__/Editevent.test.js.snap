// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditEvent /> Component should have three div element match snap 1`] = `
Array [
  <div
    className="container"
    style={
      Object {
        "paddingTop": "100px",
      }
    }
  >
    <div
      className="card card w-50 loginCard"
    >
      <div
        className="card-header dark"
      >
        <h1
          className="color"
        >
          Edit Event
        </h1>
      </div>
      <eventForm
        add={[Function]}
        centerName=""
        endDate="2018-10-23"
        formValid={false}
        getCenter={[Function]}
        minus={[Function]}
        name="tom"
        numberOfPages={
          <li
            className="page-item"
          >
            <a
              className="page-link"
            >
              Page 
              
               of 
              
            </a>
          </li>
        }
        numberOfPages1={
          <li
            className="page-item"
          >
            <a
              className="page-link"
            >
              Page 1  of 
            </a>
          </li>
        }
        onChange={[Function]}
        onSubmit={[Function]}
        purpose="for test"
        selectCenter={[Function]}
        showCenterNane={
          <h6
            className="list-group-item col-md-9"
          >
            Center Name : 
            
          </h6>
        }
        startDate="2018-10-20"
        time="10:00"
        totalPage=""
      />
    </div>
  </div>,
]
`;

exports[`<EditEvent /> Component should match component snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "paddingTop": "100px",
    }
  }
>
  <div
    className="card card w-50 loginCard"
  >
    <div
      className="card-header dark"
    >
      <h1
        className="color"
      >
        Edit Event
      </h1>
    </div>
    <div
      className="card-body"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-row"
        >
          <div
            className="form-group col-md-12"
          >
            <h5>
              <label
                htmlFor="inputEmail4"
              >
                Event Name
              </label>
            </h5>
            <input
              className="form-control form-control-lg"
              name="name"
              onChange={[Function]}
              type="text"
              value="tom"
            />
          </div>
          <div
            className="form-group col-md-6"
          >
            <h5>
              <label
                htmlFor="inputPassword4"
              >
                Start Date
              </label>
            </h5>
            <input
              className="form-control form-control-lg"
              name="startDate"
              onChange={[Function]}
              type="date"
              value="2018-10-20"
            />
          </div>
          <div
            className="form-group col-md-6"
          >
            <h5>
              <label
                htmlFor="inputPassword4"
              >
                End Date
              </label>
            </h5>
            <input
              className="form-control form-control-lg"
              name="endDate"
              onChange={[Function]}
              type="date"
              value="2018-10-23"
            />
          </div>
          <div
            className="form-group col-md-6"
          >
            <h5>
              <label
                htmlFor="inputPassword4"
              >
                Start Time
              </label>
            </h5>
            <input
              className="form-control form-control-lg"
              name="time"
              onChange={[Function]}
              type="time"
              value="10:00"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <h5>
            <label
              htmlFor="exampleFormControlTextarea1"
            >
              purpose
            </label>
          </h5>
          <textarea
            className="form-control"
            name="purpose"
            onChange={[Function]}
            placeholder="Tell people more about the event"
            rows="3"
            value="for test"
          />
        </div>
        <br />
        <div
          className="form-inline"
        >
          <button
            className="btn btn-dark btn-mm"
            data-target=".bd-example-modal-lg"
            data-toggle="modal"
            onClick={[Function]}
            style={
              Object {
                "marginRight": "20px",
              }
            }
            type="button"
          >
            Add Center
          </button>
        </div>
        <div
          aria-hidden="true"
          aria-labelledby="myLargeModalLabel"
          className="modal fade bd-example-modal-lg"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog modal-lg"
          >
            <div
              className="modal-content"
            >
              <div>
                <div
                  className="container"
                  style={
                    Object {
                      "paddingTop": "10px",
                    }
                  }
                >
                  <h1
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Centers
                  </h1>
                  <div>
                    <div
                      className="fixModel"
                    >
                      <div
                        className="card"
                        style={
                          Object {
                            "width": "12rem",
                          }
                        }
                      >
                        <img
                          alt="Card cap"
                          className="card-img-top"
                          src="llal"
                          style={
                            Object {
                              "height": "10",
                              "width": "20",
                            }
                          }
                        />
                        <div
                          className="theHeight"
                        >
                          <div
                            className="card-body text-capitalize"
                          >
                            <h6>
                              <a
                                href="/centers/29"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "color": "#2E3A41",
                                  }
                                }
                                target="_blank"
                              >
                                <strong>
                                  farm center
                                </strong>
                              </a>
                            </h6>
                            <h6>
                              No 33, yaba road lagos
                            </h6>
                            <div
                              className="scrollbar"
                            >
                              <h6>
                                <strong>
                                  Facility
                                </strong>
                              </h6>
                              <ul
                                className="list-group text-capitalize"
                              >
                                <li>
                                  car
                                </li>
                                <li>
                                  fan
                                </li>
                                <li>
                                  video
                                </li>
                              </ul>
                            </div>
                            <br />
                          </div>
                        </div>
                        <div
                          className="buttingFix"
                        >
                          <button
                            className="btn btn-primary btn-sm"
                            data-dismiss="modal"
                            onClick={[Function]}
                            type="button"
                          >
                            Select
                          </button>
                        </div>
                      </div>
                      <div
                        className="card"
                        style={
                          Object {
                            "width": "12rem",
                          }
                        }
                      >
                        <img
                          alt="Card cap"
                          className="card-img-top"
                          src="llal"
                          style={
                            Object {
                              "height": "10",
                              "width": "20",
                            }
                          }
                        />
                        <div
                          className="theHeight"
                        >
                          <div
                            className="card-body text-capitalize"
                          >
                            <h6>
                              <a
                                href="/centers/9"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "color": "#2E3A41",
                                  }
                                }
                                target="_blank"
                              >
                                <strong>
                                  farm center
                                </strong>
                              </a>
                            </h6>
                            <h6>
                              No 33, yaba road lagos
                            </h6>
                            <div
                              className="scrollbar"
                            >
                              <h6>
                                <strong>
                                  Facility
                                </strong>
                              </h6>
                              <ul
                                className="list-group text-capitalize"
                              >
                                <li>
                                  car
                                </li>
                                <li>
                                  fan
                                </li>
                                <li>
                                  video
                                </li>
                              </ul>
                            </div>
                            <br />
                          </div>
                        </div>
                        <div
                          className="buttingFix"
                        >
                          <button
                            className="btn btn-primary btn-sm"
                            data-dismiss="modal"
                            onClick={[Function]}
                            type="button"
                          >
                            Select
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                className="pagination nav justify-content-center"
              >
                <li
                  className="page-item"
                  onClick={[Function]}
                >
                  <a
                    className="page-link"
                  >
                    Previous
                  </a>
                </li>
                <li
                  className="page-item"
                >
                  <a
                    className="page-link"
                  >
                    Page 1  of 
                    2
                  </a>
                </li>
                <li
                  className="page-item"
                  onClick={[Function]}
                >
                  <a
                    className="page-link"
                  >
                    Next
                  </a>
                </li>
              </ul>
              <br />
            </div>
          </div>
        </div>
        <br />
        <br />
        <button
          className="btn btn-primary btn-lg"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<EditEvent /> Component should render initial layout of <EditEvent /> 1`] = `
Array [
  <div
    className="container"
    style={
      Object {
        "paddingTop": "100px",
      }
    }
  >
    <div
      className="card card w-50 loginCard"
    >
      <div
        className="card-header dark"
      >
        <h1
          className="color"
        >
          Edit Event
        </h1>
      </div>
      <eventForm
        add={[Function]}
        centerName=""
        endDate="2018-10-23"
        formValid={false}
        getCenter={[Function]}
        minus={[Function]}
        name="tom"
        numberOfPages={
          <li
            className="page-item"
          >
            <a
              className="page-link"
            >
              Page 
              
               of 
              
            </a>
          </li>
        }
        numberOfPages1={
          <li
            className="page-item"
          >
            <a
              className="page-link"
            >
              Page 1  of 
            </a>
          </li>
        }
        onChange={[Function]}
        onSubmit={[Function]}
        purpose="for test"
        selectCenter={[Function]}
        showCenterNane={
          <h6
            className="list-group-item col-md-9"
          >
            Center Name : 
            
          </h6>
        }
        startDate="2018-10-20"
        time="10:00"
        totalPage=""
      />
    </div>
  </div>,
]
`;
